benches/benches/bevy_ecs/schedule.rs

```diff
@@ -0,0 +1,74 @@
+use bevy_app::App;
+use bevy_ecs::prelude::*;
+use criterion::{criterion_group, criterion_main, Criterion};
+
+criterion_group!(benches, build_schedule);
+criterion_main!(benches);
+
+fn build_schedule(criterion: &mut Criterion) {
+    let mut group = criterion.benchmark_group("build_schedule");
+    group.warm_up_time(std::time::Duration::from_millis(500));
+    group.measurement_time(std::time::Duration::from_secs(15));
+
+    // Benchmark graphs of different sizes.
+    for graph_size in [100, 500, 1000, 5000] {
+        // Basic benchmark without constraints.
+        group.bench_function(format!("{graph_size}_schedule_noconstraints"), |bencher| {
+            bencher.iter(|| {
+                let mut app = App::new();
+                for i in 0..graph_size {
+                    // empty system
+                    fn sys() {}
+                    app.add_system(sys);
+                }
+            });
+        });
+        // Benchmark with constraints.
+        group.bench_function(format!("{graph_size}_schedule"), |bencher| {
+            bencher.iter(|| {
+                // empty system
+                fn sys() {}
+
+                // Use multiple different kinds of label;
+                #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, SystemLabel)]
+                struct EvenLabel(usize);
```

Comment:
Is `EvenLabel(1)` supposed to be a different label from `EvenLabel(2)`? If they are, I though labels were differentiated only by type, not by value.

Reply:
Labels are differentiated by value and type, which is how string literals work as "quick and dirty" labels. `EvenLabel(1) != EvenLabel(2)`